apps/collection/js/setup.js

```diff
@@ -32,28 +30,12 @@
 
       this.initializing = true;
 
-      var xhr = new XMLHttpRequest();
-      xhr.overrideMimeType('application/json');
-      xhr.open('GET', PRE_INSTALLED_COLLECTIONS_FILE, true);
-
-      xhr.onload = function _xhrOnLoadFile() {
-        if (!(xhr.status === 200 | xhr.status === 0)) {
-          this.onError('Unknown response when getting data.');
-          return;
-        }
-
-        try {
-          this.populate(JSON.parse(xhr.responseText));
-        } catch (ex) {
-          this.onError(ex);
-        }
-      }.bind(this);
-
-      xhr.onerror = function _xhrOnError() {
-        this.onError('file not found');
-      }.bind(this);
-
-      xhr.send();
+      LazyLoader.getJSON('js/pre_installed_collections.json')
+        .then(function(json) {
+          this.populate(JSON.parse(json));
```

Comment:
`this` does not refer to the correct object here. You need to bind the function appropriately like the existing ones did for it to work.